cpu/atmega_common/periph/uart.c

```diff
@@ -80,6 +80,11 @@ static mega_uart_t *dev[] = {
  */
 static uart_isr_ctx_t isr_ctx[UART_NUMOF];
 
+/**
+ * @brief   Allocate variable to hold transmission status, set when there
+ *          is data in UDRn or in the Transmit Shift Register
+ */
+static volatile uint8_t _tx_pending = 0;
```

Comment:
Is volatile really required ?

Reply:
AFAIK variables used inside and out of ISR should be volatile.

Reply:
ok

Reply:
> AFAIK variables used inside and out of ISR should be volatile.

@fjmolinas: No, they shouldn't. `volatile` only prevents that access are are optimized and that the order of `volatile` accesses is not changed in regard to each other. When you need to access a variable from inside and outside ISR the most readable approach is to use C11 atomics, the next best is to use `irq_disable()` ... `irq_restore()` when accessing the variable from outside of the ISR.

See https://www.kernel.org/doc/html/latest/process/volatile-considered-harmful.html and https://blog.regehr.org/archives/28